app/assets/javascripts/components/PipelineSampleReads.jsx

```diff
@@ -69,14 +69,14 @@ class PipelineSampleReads extends React.Component {
             </div>
 
             <div className="sub-title">
-              <a href="/"> {this.projectInfo.name} </a> > { this.sampleInfo.name }
+              <a href="/"> {!this.projectInfo ? 'No project to display yet' : this.projectInfo.name} </a> > { !this.sampleInfo ? 'No sample to display' : this.sampleInfo.name}
```

Comment:
Forgive me for being a complete front-end n00b, but two questions:

1.  The language inside {} -- is it Java or Ruby?

2.  Is it possible to take care of the uninitialized data case at the model level?   Where if there is no data, we crate a stub with empty fields, so that in this template we don't have to check for existence of every field.

Reply:
@boris-dimitrov, the lang inside {} is javascript
the uninitialized data case is checked for to take care of cases where null is being returned as a value from the API. 
Also, if an object with empty fields is returned, there might still need to be a way to check and show that there is "no data" returned for such fields.

Reply:
I think it would be better for the logic that decodes data (including NIL or missing data) to be in one place, rather than scattered like this.   Perhaps someone more familiar with webapp development can suggest what that place might be.   As an onbserver and occasional fixer of frontend bugs in several past projects, I can attest that scattering this type of logic tends to make it harder to assure correctness and make changes in future.